data/rulesets.js

```diff
@@ -549,7 +549,7 @@ exports.BattleFormats = {
 					}
 				}
 			}
-			if (!nonSpeedBoosted) return;
+			if (!nonSpeedBoosted || (speedBoosted !== nonSpeedBoosted && typeof speedBoosted === 'string' && typeof nonSpeedBoosted === 'string')) return;
```

Comment:
Should probably be commented "if both boost sources are Z-moves, and they're distinct"